src/FSLibrary/MissionHistoryPubnetParallelCatchupV2.fs

```diff
@@ -0,0 +1,195 @@
+            .Build()
+
+    let values =
+        use reader = new StreamReader(valuesFilePath)
+        deserializer.Deserialize<obj>(reader)
+
+    // Update the replicas value
+    let updatedValues =
+        match values with
+        | :? Dictionary<obj, obj> as dict ->
+            let mutable workerObj = null
+
+            if dict.TryGetValue(box "worker", &workerObj) then
+                match workerObj with
+                | :? Dictionary<obj, obj> as workerDict ->
+                    workerDict.[(box "stellar_core_image")] <- (box context.image)
+                    workerDict.[(box "replicas")] <- (box context.pubnetParallelCatchupNumWorkers)
+                    dict.[(box "worker")] <- box workerDict
+                | _ -> failwith "Unexpected YAML structure"
+            else
+                failwith "Unexpected YAML structure"
+
+            let mutable rangeObj = null
+
+            if dict.TryGetValue(box "range_generator", &rangeObj) then
+                match rangeObj with
+                | :? Dictionary<obj, obj> as rangeDict ->
+                    let mutable dictObj = null
+
+                    if rangeDict.TryGetValue(box "params", &dictObj) then
+                        match dictObj with
+                        | :? Dictionary<obj, obj> as paramsDict ->
+                            paramsDict.[(box "starting_ledger")] <- (box context.pubnetParallelCatchupStartingLedger)
+                            paramsDict.[(box "latest_ledger_num")] <- (box (GetLatestPubnetLedgerNumber()))
+                            rangeDict.[(box "params")] <- box paramsDict
+                        | _ -> failwith "Unexpected YAML structure"
+
+                    dict.[(box "range_generator")] <- box rangeDict
+                | _ -> failwith "Unexpected YAML structure"
+
+            dict
+        | _ ->
+            LogError "%s" (values.ToString())
+            failwith "Unexpected YAML structure"
+
+    // Serialize the updated values to a temporary file
+    let serializer =
+        SerializerBuilder()
+            .WithNamingConvention(CamelCaseNamingConvention.Instance)
+            .Build()
+
+    let tempFile = Path.GetTempFileName()
+    use writer = new StreamWriter(tempFile)
+    serializer.Serialize(writer, updatedValues)
+    writer.Flush()
+    // install the project, then delete the temporary helm values file
+    runCommand [| "helm"
+                  "install"
+                  helmReleaseName
+                  helmChartPath
+                  "--values"
+                  tempFile |]
+    |> ignore
+
+    File.Delete(tempFile)
+
+// Cleanup on exit
+let cleanup () =
+    if not cleanupCalled then
+        cleanupCalled <- true
+        LogInfo "Cleaning up resources..."
+
+        runCommand [| "helm"
+                      "uninstall"
+                      helmReleaseName |]
+        |> ignore
+
+System.AppDomain.CurrentDomain.ProcessExit.Add(fun _ -> cleanup ())
+
+Console.CancelKeyPress.Add
```

Comment:
Supercluster has a [clean](https://github.com/stellar/supercluster/blob/9bd3d8aadd8550158e0baf7b6783f7fb217014ef/src/App/Program.fs#L469-L544) command that we use in [Jenkins](https://github.com/stellar/stellar-supercluster/blob/914c4a867f2e383a7ba3ee2c5f35398d25fa249d/Jenkinsfile#L582) to make sure we teardown properly. Is there any reason to use these custom handlers instead of using that clean path?

Reply:
I don't think I understand your point - the `clean` command should clean up `StatefulSet` as well right?

Reply:
The jenkins clean is a separate command that cleans up all the artifacts of the entire pipeline, but not after each step (mission) correct?
Here the cleanup function makes sure the clean up happens after the parallel catchup, otherwise the statefulset resources will be occupied for the missions after it.

Reply:
Ah that's right. We still do cleanup (I think by calling `StellarFormations` `Cleanup` method after a mission completes. Wouldn't it make sense to put this code in there? We already clean up pods between missions without calling the `clean` command.

Reply:
I'm not using a `StellarFormation` any more, everything is directly managed under Helm now.